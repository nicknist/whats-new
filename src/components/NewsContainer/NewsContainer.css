.headline {
  height: 100px;
  width: 100px;
  margin: 7% 0%;
}

main {
  overflow: scroll;
  display: flex;
  flex-flow: wrap;
}

article {
  width: 48%;
  background-color: #F9F68A;
  margin: 1%;
  padding: 3%;
}

a {
  background-color: #55A5FF;
  color: midnightblue;
  display: block;
  margin-top: 3%;
}
