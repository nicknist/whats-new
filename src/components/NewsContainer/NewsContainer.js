import React from 'react';
import './NewsContainer.css';
import NewsArticle from './../NewsArticle/NewsArticle'
import PropTypes from 'prop-types'

// NEWSCONTAINER COMPONENT CODE GOES HERE

const NewsContainer = ({ data }) => {
  const newsArticles = data.map(article => {
    return <NewsArticle
            key={article.url}
            articleInfo={article}
          />
  })

  return (
    <main className="news-container">
      {newsArticles}
    </main>
  )
}

NewsContainer.propTypes = {
  data: PropTypes.array
}

export default NewsContainer;
