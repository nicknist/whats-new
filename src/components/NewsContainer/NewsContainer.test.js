import React from 'react';
import NewsContainer from './NewsContainer';
import { shallow } from 'enzyme';

describe('NewsContainer', () => {
  it('should create to area for all the articles to be in', () => {
    const mockData = [
      {
        id: 1,
        headline: 'Mock Headline 1',
        img: 'fakeurl.com',
        description: 'Fake fake stuff.',
        url: 'anotherfakeurl.net'
      },
      {
        id: 2,
        headline: 'Mock Headline 2',
        img: 'fakeurl2.com',
        description: 'More fake stuff',
        url: 'anotherfakeurl2.net'
      }
    ];
    const wrapper = shallow(<NewsContainer data={mockData}/>)

    expect(wrapper).toMatchSnapshot();
  })
})
