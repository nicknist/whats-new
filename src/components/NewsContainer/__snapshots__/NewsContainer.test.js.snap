// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsContainer should create to area for all the articles to be in 1`] = `
<main
  className="news-container"
>
  <NewsArticle
    articleInfo={
      Object {
        "description": "Fake fake stuff.",
        "headline": "Mock Headline 1",
        "id": 1,
        "img": "fakeurl.com",
        "url": "anotherfakeurl.net",
      }
    }
    key="anotherfakeurl.net"
  />
  <NewsArticle
    articleInfo={
      Object {
        "description": "More fake stuff",
        "headline": "Mock Headline 2",
        "id": 2,
        "img": "fakeurl2.com",
        "url": "anotherfakeurl2.net",
      }
    }
    key="anotherfakeurl2.net"
  />
</main>
`;
